src/core/data_loader/dedupe.rs

```diff
@@ -185,4 +229,59 @@ mod tests {
             "compute_value was called more than once"
         );
     }
+
+    #[tokio::test]
+    async fn test_hanging_after_dropped() {
+        let cache = Arc::new(Dedupe::<u64, ()>::new(100, true));
+
+        let task = cache.dedupe(&1, move || async move {
+            sleep(Duration::from_millis(100)).await;
+        });
+
+        // drops the task since the underlying sleep timeout is higher than the
+        // timeout here
+
+        timeout_at(Instant::now() + Duration::from_millis(10), task)
+            .await
+            .expect_err("Should throw timeout error");
+
+        cache
+            .dedupe(&1, move || async move {
+                sleep(Duration::from_millis(100)).await;
+            })
+            .await;
+    }
+
+    #[tokio::test]
+    async fn test_hanging_dropped_while_in_use() {
```

Comment:
This is is better than the `test_hanging_after_dropped` test. Can we drop the above one?